nslsii/__init__.py

```diff
@@ -320,38 +332,45 @@ def configure_ipython_exc_logging(exception_logger, ipython, rotate_file_size=10
         at the time configure_ipython_exc_logging() is called, if there exists a log file
         with size in bytes greater than or equal to rotate_file_size, the existing file will
         be renamed and a new log file will be created
+    appdirs_appname: str
+        appname passed to appdirs.user_log_dir(); use the default for production,
+        set to something else for testing
 
     Returns
     -------
-    bluesky_ipython_log_file_path: str
+    bluesky_ipython_log_file_path: Path
         log file path
 
     """
     ipython.set_custom_exc((BaseException,), exception_logger)
 
     if "BLUESKY_IPYTHON_LOG_FILE" in os.environ:
+        bluesky_ipython_log_file_path = Path(os.environ["BLUESKY_IPYTHON_LOG_FILE"])
         print(
             "bluesky ipython log file configured from environment"
             " variable BLUESKY_IPYTHON_LOG_FILE",
             file=sys.stderr,
         )
-        bluesky_ipython_log_file_path = os.environ["BLUESKY_IPYTHON_LOG_FILE"]
     else:
+        bluesky_ipython_log_file_path = Path(
+            appdirs.user_log_dir(appname=appdirs_appname)
+        ) / Path("bluesky_ipython.log")
         print(
             "environment variable BLUESKY_IPYTHON_LOG_FILE is not set,"
             " using default file path",
             file=sys.stderr,
         )
-        bluesky_ipython_log_file_path = "/var/log/bluesky/bluesky_ipython.log"
-    # before starting ipython logging check the size of the ipython log ile
+    if not bluesky_ipython_log_file_path.parent.exists():
+        bluesky_ipython_log_file_path.parent.mkdir(parents=True, exist_ok=True)
+    # before starting ipython logging check the size of the ipython log file
     # if the ipython log file has grown large make a copy and start a new one
     # if a previous copy exists just overwrite it
     if (
-        os.path.exists(bluesky_ipython_log_file_path)
+        bluesky_ipython_log_file_path.exists()
         and os.path.getsize(bluesky_ipython_log_file_path) >= rotate_file_size
     ):
-        os.replace(
-            bluesky_ipython_log_file_path, bluesky_ipython_log_file_path + ".old"
+        bluesky_ipython_log_file_path.rename(
+            str(bluesky_ipython_log_file_path) + ".old"
```

Comment:
Why not to use `TimedRotatingFileHandler` here?

Reply:
We can only submit an exception logging *function* to ipython. We can't mess with the logging internals (as far as I can tell).

Reply:
I should have mentioned the `bluesky_ipython_log_file_path` gets *all* ipython output, so it is not really a target for `logging`.

Reply:
Now that I think about it the name `configure_ipython_exc_logging` is *wrong*, since it configures both normal output and exception logging. I am changing it to `configure_ipython_logging` and adding some clarification to the docstring.